Bugsnag/Configuration/BugsnagConfiguration.m

```diff
@@ -216,6 +217,11 @@ - (instancetype _Nonnull)initWithApiKey:(NSString *_Nonnull)apiKey
     #elif BSG_PLATFORM_OSX
         _appType = @"macOS";
     #endif
+
+    // Quick fix to get version info working, but the rest of the default values are still not persisted.
+    [self setAppVersion:_appVersion];
```

Comment:
The only other fields I can see in the same category are `appType` (which won't change between launches) and `releaseStage` (which is possibly but unlikely to change). The other fields all control functionality that we'd want to use the value from the current (not the persisted) config.

Suggest you remove the direct setting of the fields from lines 175/176 and simply call the setters for app version, bundle version, app type and release stage here.

I don't see why this is a quick fix - this seems legitimate.

Reply:
Looks like Tom made an unresolved point here - the same behavior as was problematic for `appVersion` and `bundleVersion` is also being used for release stage, which can be considered versioning info. It looks like it also needs to be propagated into config.